python/pyarrow/_dataset.pyx

```diff
@@ -881,10 +881,15 @@ cdef class RowGroupInfo:
             name = frombytes(c_statistics.type.get().field(i).get().name())
             c_minmax = <CStructScalar*> c_statistics.value[i].get()
 
-            statistics[name] = {
-                'min': pyarrow_wrap_scalar(c_minmax.value[0]).as_py(),
-                'max': pyarrow_wrap_scalar(c_minmax.value[1]).as_py(),
-            }
+            try:
+                statistics[name] = {
+                    'min': pyarrow_wrap_scalar(c_minmax.value[0]).as_py(),
+                    'max': pyarrow_wrap_scalar(c_minmax.value[1]).as_py(),
+                }
+            except ValueError:
+                # Don't treat failure to parse/convert a single Scalar as a
+                # failure. The min/max will simply be missing for this field.
```

Comment:
Should we add a test case for this? IIUC this will only arise when a parquet file has statistics of a type which `pyarrow_wrap_scalar` doesn't support; are we sure we want to hide that error?

Reply:
See also related question above: https://github.com/apache/arrow/pull/7623#discussion_r449647463

I am not sure we *can* test it right now, as all scalars are supported in Python. 

I suppose the rationale is that as a user experience it can be better to still return the statistics of all other columns, instead of raising due to one column not being recognized. 
On the other hand, it indeed suppresses an error that would be useful to see (for us developers).

Reply:
IMHO it'd be better to raise an exception. If conversion fails in the future for a statistics scalar which we don't yet support then recognizing that issue will be easier than figuring out why the statistics disappear.

Reply:
We should support Scalars of all types, so it should not be possible for this to error, and if it does, it's a bug. IMO we should remove the try/except and move on.

Reply:
Okido, removed!